ci.md

```diff
@@ -0,0 +1,201 @@
+# Ignition Continuous Integration
+
+When developers push code changes into Ignition, a series of automated tests
+run for various platforms. This process is called Continuous Integration, or CI.
+This document covers all the CI jobs that are available for Ignition and how to
+interpret their results.
+
+## Types of checks
+
+The main type of check that is performed by CI is compiling the code and running
+all automated tests. Some jobs may also run linters and other static checkers.
+Finally, we have checks that add labels, check DCO and coverage.
+
+All checks are visible at the bottom of a pull request.
+
+![PR checks](images/PR_checks.png)
+
+We use two platforms to run CI, each of them running different builds.
+
+### Jenkins server
+
+Our Jenkins server is located at https://build.osrfoundation.org/. It runs builds
+for all supported operating systems. It also runs an ABI checker for all stable
+branches.
+
+Each pull request to a stable or `main` branch will trigger:
+
+* `<library>-ci-pr_any-ubuntu-auto-amd64`: Build and test on Ubuntu Linux
+    * Builds are run inside Docker containers for Bionic or Focal.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-homebrew-amd64`: Build and test on macOS
+    * Dependencies are installed from homebrew.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-windows*-amd64`: Build and test on Windows
+    * External dependencies are installed from vcpkg
+    * Library being tested and its Ignition dependencies are compiled using colcon.
+
+Pull requests to stable branches also trigger:
+
+* `<library>-abichecker-any_to_any-ubuntu_auto-amd64`: Run ABI checker
+
+### GitHub Actions
```

Comment:
It would be nice to include a summary/table at the end of this section that lists what checks are different between Jenkins and GitHub actions.

Reply:
I turned the entire section into a table :smirk: